### PR TITLE
Make chunked mode in salt-cp optional

### LINKED ISSUE #42627: salt-cp no longer works.  Was working a few months back.
Copying a file from salt-master to salt-minion using salt-cp.  File is only 5 bytes.

# salt-call test.versions_report

local:
    Salt Version:
               Salt: 2016.11.6
     
    Dependency Versions:
               cffi: Not Installed
           cherrypy: 5.6.0
           dateutil: Not Installed
          docker-py: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.8.1
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: Not Installed
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.4.6
       mysql-python: Not Installed
          pycparser: Not Installed
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 2.6.9 (unknown, Sep  1 2016, 23:34:36)
       python-gnupg: Not Installed
             PyYAML: 3.11
              PyZMQ: 14.5.0
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.2.1
                ZMQ: 4.0.5
     
    System Versions:
               dist:   
            machine: x86_64
            release: 4.4.11-23.53.amzn1.x86_64
             system: Linux
            version: Not Installed

# ls -la test.txt
-rw-r--r--  1 root     root        5 Jul 29 01:38 test.txt

# salt-cp 'minion_host' test.txt /tmp/

----------
    /tmp/test.txt:
        Passed invalid arguments to cp.recv: recv() takes exactly 2 arguments (5 given)
        
            Used with salt-cp, pass the files dict, and the destination.
        
            This function receives small fast copy files from the master via salt-cp.
            It does not work via the CLI.

### PR DESCRIPTION
**Pull request by DmitryKuzmenko**
### What does this PR do?
Make chunked mode in salt-cp optional, disabled by default.
It could be enabled with `--chunked` or `-C` cmdline flag.

### What issues does this PR fix or reference?
Fixes #42627 

### Tests written?
No

### CODE REVIEW

File: doc/ref/cli/salt-cp.rst [hunk 1]
```diff
@@ -39,6 +39,12 @@ specified target expression.
     desitination will be assumed to be a directory. Finally, recursion is now
     supported, allowing for entire directories to be copied.
 
+.. versionchanged:: 2016.11.7,Nitrogen
```
Comment (rallytime) [inline doc/ref/cli/salt-cp.rst]: This versionchanged tag needs to be updated: `Nitrogen` should be `2017.7.2`

File: doc/ref/cli/salt-cp.rst [hunk 2]
```diff
@@ -39,6 +39,12 @@ specified target expression.
...
+    Returned tha old mode when salt-cp copies all files in one command back.
+    The new chunked mode is available with the new ``-C``, ``--cunked`` argument.
```
Comment (rallytime) [inline doc/ref/cli/salt-cp.rst]: Minor spelling fix: `--cunked` should be `--chunked`

File: doc/ref/cli/salt-cp.rst [hunk 3]
```diff
@@ -56,9 +62,16 @@ Options
 .. include:: _includes/target-selection.rst
 
 
+.. option:: -C, --chunked
+
+    Use new chunked mode to copy files. This mode supports large files, recursive
+    directories copying and compression.
+
+    .. versionadded:: 2016.11.7,Nitrogen
```
Comment (rallytime) [inline doc/ref/cli/salt-cp.rst]: Same version change here: `Nitrogen` -> `2017.7.2`

File: doc/ref/cli/salt-cp.rst [hunk 4]
```diff
@@ -39,6 +39,12 @@ specified target expression.
     desitination will be assumed to be a directory. Finally, recursion is now
     supported, allowing for entire directories to be copied.
 
+.. versionchanged:: 2016.11.7,Nitrogen
+    Returned tha old mode when salt-cp copies all files in one command back.
```
Comment (rallytime) [inline doc/ref/cli/salt-cp.rst]: Spelling: `tha` should be `the`.

**Review by rallytime — CHANGES_REQUESTED** [review]
@DmitryKuzmenko This looks good to me. I just had a couple of small spelling and versioning items to fix here.

@terminalmage Should review this as well.

**Comment by DmitryKuzmenko** [comment]
Fixed. Thank you @rallytime !

**Review by rallytime — APPROVED** [review]

**Review by terminalmage — APPROVED** [review]

File: doc/ref/cli/salt-cp.rst [hunk 5]
```diff
@@ -39,6 +39,12 @@ specified target expression.
     desitination will be assumed to be a directory. Finally, recursion is now
     supported, allowing for entire directories to be copied.
 
+.. versionchanged:: 2016.11.7,2017.7.2
```
Comment (terminalmage) [inline doc/ref/cli/salt-cp.rst]: The wording here could be improved. How about this?

```rst
.. versionchanged:: 2016.11.7,2017.7.2
    Reverted back to the old copy mode to preserve backward compatibility. The
    new functionality added in 2016.6.6 and 2017.7.0 is now available using the
    ``-C`` or ``--chunked`` CLI arguments. Note that compression, recursive
    copying, and support for copying large files is only available in chunked
    mode.
```

**Review by terminalmage — APPROVED** [review]
This looks good. There's only one change I'd make, and it's in the documentaton. We can always get that changed in a separate PR though.

File: doc/ref/cli/salt-cp.rst (hunk 5 above)
Comment (DmitryKuzmenko) [inline doc/ref/cli/salt-cp.rst]: Thank you @terminalmage .
I've updated the doc.

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by garethgreenaway** [comment]
Go Go Jenkins!

**Pull request merged**